lib/rust/metamodel/src/meta/mod.rs

```diff
@@ -0,0 +1,388 @@
+//! An abstracted representation of a data model.
+//!
+//! This is used in translation from Rust to Java to:
+//! - Decouple the complexities of the source language from those of the target language.
+//! - Provide a simple representation to support analysis and translation.
+
+#[cfg(feature = "graphviz")]
+pub mod graphviz;
+pub mod serialization;
+pub mod transform;
+
+use crate::data_structures::VecMap;
+use derive_more::Index;
+use derive_more::IndexMut;
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+
+
+
+// ==============================
+// === Type Parameterizations ===
+// ==============================
+
+/// Globally unique, stable identifier for a `Field`.
+pub type FieldId = crate::data_structures::Id<Field>;
+
+/// Identfies a type within a `TypeGraph`.
+pub type TypeId = crate::data_structures::vecmap::Key<Type>;
+/// Identfies an unbound type within a `TypeGraph`.
+pub type UnboundTypeId = crate::data_structures::vecmap::UnboundKey<Type>;
+
+
+
+// ======================
+// === Datatype Types ===
+// ======================
+
+/// A datatype.
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct Type {
```

Comment:
I'm reading this file from top to bottom. I'm on the line 40 and I still dont understand what it defines. On line 40 there is a "Type" which is "a datatype". There is no explanation in what language this is and what it represents. This is probably a Rust data type, but it is not explained anywhere explicitly.

Edit: it seems this is not rust data type. I'm even more confused right now. If this is a Java AST, why it is not defined in a java/syntax file?

Reply:
I have tried to clarify it. The new crates docs provide a lot of context that should help here.